### DISCUSSION TITLE
PostgresML is 8-40x faster than Python HTTP microservices - PostgresML

### DISCUSSION THREAD

**Discussion by giscus** [postgresml/postgresml#422] [Announcements]
# PostgresML is 8-40x faster than Python HTTP microservices - PostgresML

PostgresML v2.0 outperforms Python HTTP microservices by a factor of 8 in throughput, latency, and memory utilization.

https://postgresml.org/blog/postgresml-is-8x-faster-than-python-http-microservices/

**Comment by jfmatth** [comment]
I'm not an ML guy, but love to hear the Postgres ecosystem is doing great here!

**Comment by arvsr1988** [comment]
This is not a fair comparison. You need to serve the postgres query over a http server just like the python application.

**Reply by montanalow** [reply]
I think the point the post illustrates beyond the straight benchmarks, is that you actually don't need an http service to access ML with PostgresML. Connectivity comes from PostgreSQL instead. It's a completely different stack with a completely different performance profile.
